File: base/src/main/java/com/base/broadcast/InternetBroadCastReceiver.java
package com.base.broadcast;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.base.Utils;
import com.base.interfaces.ConnectivityListener;

/**
 * Created by amit on 20/2/17.
 */

public class InternetBroadCastReceiver extends BroadcastReceiver {
    public ConnectivityListener connectivityListener;

    public void addCallback(ConnectivityListener listener) {
        this.connectivityListener = listener;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        boolean isConnected = Utils.isInternetConnected(context);
        if (connectivityListener != null) {
            connectivityListener.onConnectivityChange(isConnected);
        }
    }
}

File: activity/src/main/java/com/common/broadcast/InternetBroadCastReceiver.kt
package com.common.broadcast

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.common.Utils
import com.common.interfaces.ConnectivityListener


/**
 * Created by amit on 20/2/17.
 */

class InternetBroadCastReceiver : BroadcastReceiver() {
    private var connectivityListener: ConnectivityListener? = null

    fun addCallback(listener: ConnectivityListener) {
        this.connectivityListener = listener
    }

    override fun onReceive(context: Context, intent: Intent) {
        val isConnected = Utils.isInternetConnected(context)
        connectivityListener?.onConnectivityChange(isConnected)
    }
}
